csp/impl/types/instantiation_type_resolver.py

```diff
@@ -386,7 +386,17 @@ def _add_scalar_value(self, arg, in_out_def):
     def _is_scalar_value_matching_spec(self, inp_def_type, arg):
         if inp_def_type is typing.Any:
             return True
-        if UpcastRegistry.instance().resolve_type(inp_def_type, type(arg), raise_on_error=False) is inp_def_type:
+        if inp_def_type is typing.Callable or (
```

Comment:
Can we add `CspTypeUtils.is_callable` to encapsulate this complex set of conditions?